src/mlpack/methods/edge_boxes/feature_extraction_impl.hpp

```diff
@@ -0,0 +1,903 @@
+/**
+ * @file feature_extraction_impl.hpp
+ * @author Nilay Jain
+ *
+ * Implementation of feature extraction methods.
+ */
+#ifndef MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_IMPL_HPP
+#define MLPACK_METHODS_EDGE_BOXES_STRUCTURED_TREE_IMPL_HPP
+
+
+#include "feature_extraction.hpp"
+#include <map>
+
+namespace mlpack {
+namespace structured_tree {
+
+template<typename MatType, typename CubeType>
+StructuredForests<MatType, CubeType>::
+StructuredForests(const std::map<std::string, int>& inMap)
```

Comment:
Since you need to copy the inMap no matter what, maybe you could pass by value, this is more effecient([Want performance](http://stackoverflow.com/questions/21605579/how-true-is-want-speed-pass-by-value)? pass by value)

`StructuredForests(std::map<std::string, int> inMap) : options(std::move(inMap)) {}`

I use initialization list to initialize the data member.The class/struct of c++ do the initialization task at initialization list, not constructor. If you assign the inMap to options in constructor, it is "assign" but not "initialize".